src/main/scala/ir/dsl/DSL.scala

```diff
@@ -92,31 +92,33 @@ def R(i: Int): Register = Register(s"R$i", 64)
 
 def bv_t(i: Int) = BitVecType(i)
 
+case class Resolver(program: Program) {
```

Comment:
i wonder if it would be good to make this private. otherwise, add a comment warning the user how to use it correctly wrt making a new Resolver when blocks/procs are changed.

_also,_ i'm unsure of the Resolver name, since we already use resolver for a different thing in this file (the lambda continuation in the `resolve` methods). but the best i can come up with is CachedProcedureAndBlocks which is.. not great

_also_, i think this bookkeeping should eventually be moved into Program/Procedure. i'm sure it's useful elsewhere as well.